app/src/main/java/org/kinecosystem/kinit/viewmodel/earn/QuizQuestionViewModel.kt

```diff
@@ -0,0 +1,160 @@
+package org.kinecosystem.kinit.viewmodel.earn
+
+import android.databinding.ObservableBoolean
+import android.databinding.ObservableInt
+import org.kinecosystem.kinit.KinitApplication
+import org.kinecosystem.kinit.analytics.Analytics
+import org.kinecosystem.kinit.analytics.Events
+import org.kinecosystem.kinit.model.earn.*
+import org.kinecosystem.kinit.repository.TasksRepository
+import org.kinecosystem.kinit.util.Scheduler
+import org.kinecosystem.kinit.view.adapter.AnswersListAdapter
+import org.kinecosystem.kinit.view.customView.QuizAnswerView
+import org.kinecosystem.kinit.view.earn.QuestionnaireActions
+import javax.inject.Inject
+
+open class QuizQuestionViewModel(private var questionIndex: Int,
+                                 private val questionnaireActions: QuestionnaireActions) {
+
+
+    private val delay_show_correct_answer = 500L
+    @Inject
+    lateinit var scheduler: Scheduler
+    @Inject
+    lateinit var taskRepository: TasksRepository
+    @Inject
+    lateinit var analytics: Analytics
+
+    private var questionObj: Question? = null
+    private var questionAnswered = false
+    var chosenAnswersIndexes: MutableList<Int> = mutableListOf()
+    var chosenAnswers: MutableList<String> = mutableListOf()
+    var chosenAnswersCount: ObservableInt = ObservableInt(chosenAnswers.size)
+    var question: String?
+    var answers: List<Answer>?
+    private var quizData: QuizData
+    var reward: Int
+    var clickable = ObservableBoolean(true)
+    var showCorrect: ObservableBoolean = ObservableBoolean(false)
+
+    var selectionListener: QuizAnswerView.OnSelectionListener = object : QuizAnswerView.OnSelectionListener {
+        override fun onAnimComplete() {
+            questionnaireActions.next()
+        }
+
+        override fun onAnswered(answer: Answer?) {
+            clickable.set(false)
+            answer?.let {
```

Comment:
Why Answer's id is an optional field?

Reply:
answer has a custom set hence it cant be lateinit

Reply:
if theres error in the data taken from server we dont want to crash in the retrofit parsing
so data class from server might have null and we check if its valid